squad-track-duration

```diff
@@ -313,60 +302,126 @@ def run():
         ["created_at", "git_describe", "device", "build_name_device", "build_name"]
-    )["boottime"].mean()
+    )
+
+    mean_boottimes2 = df_grouping2["boottime"].mean()
+
+    # Convert the Series object back to a DataFrame then sort values first by
+    # build_name_device, then by created_at. This will make the graph legend
+    # alphabetised while also ensuring the dates for each line are ordered by
+    # created_at so the graph's lines will be drawn correctly.
+    mean_boottimes2 = mean_boottimes2.reset_index().sort_values(
+        by=["build_name_device", "created_at"]
+    )
 
-    # Convert the Series object back to a DataFrame then sort by the created_at
-    # and build_name_device
-    dfp = dfp.reset_index().sort_values(by=["created_at", "build_name_device"])
+    logger.debug(mean_boottimes2.info())
+    logger.debug(mean_boottimes2)
 
-    # Filter by results from the specified build names
-    dfp = dfp[dfp["build_name"].str.endswith(args.build_name.split("-")[-1])]
-    logger.debug(dfp.info())
-    logger.debug(dfp)
+    # Calculate how many boottimes we averaged over per device
+    count_per_device2 = (
+        df_grouping2["boottime"].count().groupby("build_name_device").sum()
+    )
+    count_per_device2 = count_per_device2.reset_index().rename(
+        columns={"boottime": col_name_boottime_count}
+    )
+
+    # Create a new column with the name and count, then stick together the
+    # counts and the averages
+    count_per_device2["build_name_device_count"] = (
+        count_per_device2.build_name_device
+        + " ("
+        + count_per_device2[col_name_boottime_count].astype(str)
+        + ")"
+    )
+    mean_boottimes2 = mean_boottimes2.merge(
+        count_per_device2, on="build_name_device", how="inner", suffixes=("_1", "_2")
+    )
 
     # Create the figure for this visualisation
     figure_colletion.append(
         MetaFigure(
             px.line(
-                dfp,
+                mean_boottimes2,
                 x="created_at",
                 y="boottime",
-                color="build_name_device",
+                color="build_name_device_count",
                 markers=True,
-                labels={"build_name_device": "Build name - device"},
+                labels={"build_name_device_count": "Build name - device (# boottimes)"},
+            )
+            .update_xaxes(
+                tickvals=mean_boottimes2["created_at"],
+                ticktext=mean_boottimes2["git_describe"],
             )
-            .update_xaxes(tickvals=dft["created_at"], ticktext=dft["git_describe"])
             .update_layout(xaxis_title="Version", yaxis_title="Boot time"),
             f"Line graph, {args.build_name.split('-')[-1]}",
-            f"This line graph, is generated from \"{args.build_name.split('-')[-1]}\".",
+            f"This line graph, is generated from an average over {count_per_device2[col_name_boottime_count].sum()} boottimes for \"{args.build_name.split('-')[-1]}\".",
```

Comment:
So the average isn't over this, different per device right.
Can we reformulate it to mach that?
Maybe something like this?
"This line graph, is generated form an average (or a mean) over number of boots, the number of boots is presented in the  'Build name - device (mean over number of boots)' in the line graph."

Reply:
Updated the text a bit (and for the other comment, too) - let me know what you think!